contrib/mesos/pkg/scheduler/podtask/pod_task.go

```diff
@@ -170,10 +198,7 @@ func (t *T) Has(f FlagType) (exists bool) {
 	return
 }
 
-func New(ctx api.Context, id string, pod api.Pod, executor *mesos.ExecutorInfo) (*T, error) {
-	if executor == nil {
-		return nil, fmt.Errorf("illegal argument: executor was nil")
-	}
+func New(ctx api.Context, id string, pod *api.Pod) (*T, error) {
```

Comment:
I prefer to have that behavior explicit in the `*pod` expression. As far as I see, it's not less shared now than it was before.